src/components/LogIn/LogIn.js

```diff
@@ -0,0 +1,258 @@
+import React from "react";
+import { TouchableOpacity, Text, View, TextInput, Alert } from "react-native";
+import { environment } from "../../environment/environment";
+import styles from "../../global/global-styles.js";
+import globalConstant from "../../global/global-constant.js";
+import { Formik } from "formik";
+import * as yup from "yup";
+import * as SecureStore from "expo-secure-store";
+/**
+ * @description This class is used to display the log in form
+ * where the user can log in or reset their password
+ */
+function LogIn() {
+  const token = "";
+  //use to add validation to the form by using yup
+  const loginValidationSchema = yup.object().shape({
+    email: yup
+      .string()
+      .email("Please enter valid email")
+      .required("Email is required"),
+    password: yup.string().required("Password is required"),
+  });
+
+  //use to change the error message displayed to the user
+  const [error, setError] = React.useState(null);
+
+  /**
+   * async function used to sign in the user
+   * @param {*} data is the values entered by the user
+   */
+  async function onSignInTap(data) {
+    const url = environment["authHost"] + "api/user/post/login";
+    try {
+      const response = await fetch(url, {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify(data),
+      });
+      let responseJSON = await response.json();
+      if (responseJSON["status"] == "valid") {
+        await SecureStore.setItemAsync("jwt", responseJSON["data"]["jwt"]);
+        updateTokenInDatabase();
+      } else {
+        if (responseJSON["data"]) {
+          setError(responseJSON["data"]["message"]);
+        } else {
+          setError(responseJSON["error"]["code"]);
```

Comment:
you should check the code and send a more redable message

Reply:
I didn't find a specific error code for this. The 2 possible errors that I found were already caught in the responseJSON["data"]["message"].